tables/leaf.py

```diff
@@ -92,6 +105,46 @@ def calc_chunksize(expected_mb: int) -> int:
     return chunksize * 8
 
 
+class ChunkInfo(NamedTuple):
+    """Information about storage for a given chunk.
+
+    It may also refer to a chunk which is within the dataset's shape but that
+    does not exist in storage, i.e. a missing chunk.
+
+    An instance of this named tuple class contains the following information,
+    in field order:
+
+    .. attribute:: start
+
+        The coordinates in dataset items where the chunk starts, a tuple of
+        integers with the same rank as the dataset.  These coordinates are
+        always aligned with chunk boundaries.  Also present for missing
+        chunks.
+
+    .. attribute:: filter_mask
+
+        An integer where each active bit signals that the filter in its
+        position in the pipeline was disabled when storing the chunk.  For
+        instance, ``0b10`` disables shuffling, ``0b100`` disables szip, and so
+        on.  ``None`` for missing chunks.
+
+    .. attribute:: offset
+
+        An integer which indicates the offset in bytes of chunk data as it
+        exists in storage.  ``None`` for missing chunks.
+
+    .. attribute:: size
+
+        An integer which indicates the size in bytes of chunk data as it
+        exists in storage.  ``None`` for missing chunks.
+
+    """
+    start: Union[tuple[int, ...], None]
+    filter_mask: Union[int, None]
+    offset: Union[int, None]
+    size: Union[int, None]
```

Comment:
I think that we dropped support to Python 3.9.
You can use the syntax `int | None` instead of `Union[int, None]` if you want.

Reply:
Thanks! I see no other uses of the newer syntax in code, maybe we can leave `Union` for consistency, and upgrade annotations later on.